java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfig.java

```diff
@@ -53,6 +53,17 @@
   private Map<Integer, JdbcFieldInfo> arraySubTypesByColumnIndex;
   private Map<String, JdbcFieldInfo> arraySubTypesByColumnName;
 
+  public static final int DEFAULT_PARTIAL_LIMIT = -1;
```

Comment:
you should distinguish between DEFAULT which should be something like 512, 1024 or 2048 and NO_LIMIT which should be -1.